libtft.py

```diff
@@ -138,4 +138,35 @@ def repo_create(path):
         config = repo_default_config()
         config.write(f)
 
-    return repo
\ No newline at end of file
+    return repo
+
+def take_parent_dir(path):
```

Comment:
I prefer the recursive solution suggested by the guide. You also mentioned this is not working for all paths.
Readability >>>